src/core/connection.ts

```diff
@@ -138,16 +138,20 @@ export class Connection extends EventEmitter {
       return this.getKeys(match, count, keysAry, parseInt(newCursor));
     }
 
+
     this.emit(
       "error",
       new Error(
         "You must establish a connection to redis before running the getKeys command."
       )
     );
+
+    return null;
   }
 
   end() {
-    Object.keys(this.listeners).forEach((eventName) => {
+    Object.keys(this.listeners).forEach((eventName:string) => {
+      // @ts-ignore
```

Comment:
must not add ts-ignore without comment! either remove or add comment what rule is being ignored.